include/triton/core/tritonserver.h

```diff
@@ -1703,6 +1703,37 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error*
 TRITONSERVER_ServerOptionsSetResponseCacheByteSize(
     TRITONSERVER_ServerOptions* options, uint64_t size);
 
+/// Set the address of the redis (or redis cluster) instance
+/// to be used for the request response cache
+///
+/// \param options The server options object.
+/// \param address The Redis cache address
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_ServerOptionsSetResponseCacheAddress(
+    TRITONSERVER_ServerOptions* options, char* address);
+
+/// Set the address of the redis (or redis cluster) instance
+/// to be used for the request response cache
+///
+/// \param options The server options object.
+/// \param password The password to use for communication with Redis.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_ServerOptionsSetResponseCachePassword(
+    TRITONSERVER_ServerOptions* options, char* password);
+
+/// Set the username of the redis (or redis cluster) instance
+/// to be used for the request response cache
+///
+/// \param options The server options object.
+/// \param username The username to use for communication with Redis
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_ServerOptionsSetResponseCacheUsername(
+    TRITONSERVER_ServerOptions* options, char* username);
+
+
```

Comment:
Moving forward with a more generic API for creating/configuring/using a cache (of either type), we may want to instead make use of `TRITONSERVER_Parameter** parameters` to pass in arbitrary key-value pairs for configuring either type of cache and some kind of `TRITONSERVER_CacheKind kind` for the various APIs to decide what to do for each.

ex: Redis expects username/password/address and Local expects  cache_size, etc.

CC @guanluo

Reply:
This is exactly the plan. Use a JSON file to specify these attributes and then have an enum that controls the change in cache type.

Reply:
Reading this again and reading the code again, I now see what you meant. This is very doable.